File: cpp/duck/Duck/Duck.cpp
#include "Duck.h"

#include <utility>

Duck::Duck(string name) : name(std::move(name)) {}

void Duck::quack() {
    cout << name << " is quacking" << endl;
}

void Duck::swim() {
    cout << name << " is swimming" << endl;
}

void Duck::display() {
    cout << "I am a " << name << endl;
}

File: golang/duck/client/Duck.go
package client

import (
	fb "duck/behavior"
	"fmt"
)

var FlyB fb.FlyBehavior

type Duck struct { Name string }

func (duck Duck) Display() { fmt.Println("I am a " + duck.Name) }

func (duck Duck) Swim() { fmt.Println(duck.Name + " is swimming") }

func (duck Duck) PerformFly() {
	FlyB.Fly()
}
